components/bl-mapbox/src/helpers.js

```diff
@@ -199,7 +199,7 @@ export const useMarkers = (markers, mapRef, onMarkerClick) => {
         popup.on('open', () => {
           const coordinates = { lat: markerItem.coordinates.lat, lng: markerItem.coordinates.lng };
 
-          onMarkerClick({ coordinates, description: description || '' });
+          onMarkerClick({ coordinates, description: description || '', markerData });
         });
 
         if (description) {
```

Comment:
logic below

```
    if (polygon.outlineWidth > 0) {
        map.removeLayer(`${ polygon.id }-outline`);
      }
      map.removeLayer(`${ polygon.id }-layer`);
      map.removeSource(polygon.id);
      ```
      
Should be done inside one Map class method that accept `polygon`

Functions that already created is useless(removeLayer, removeSource)